### PR TITLE
fix: Update and sync example Azure pipeline with code from ci-cd-references repo

### PR DESCRIPTION
**Pull request by ananthkamath**

### CODE REVIEW

**Review by endor-matt — APPROVED** [review]

**Comment by scott-es** [comment]
just FYI it looks like each pull request includes the commits from the prior one, what I'm saying is merging this one will bring in the 4 commits from PR 1-4 so once that happens I dont think the other 3 matter anymore.

**Comment by ananthkamath** [comment]
> just FYI it looks like each pull request includes the commits from the prior one, what I'm saying is merging this one will bring in the 4 commits from PR 1-4 so once that happens I dont think the other 3 matter anymore.

@scott-es I was planning to rebase every PR, after the older one gets merged in. Wanted to keep changes limited in every PR.

**Comment by scott-es** [comment]
a few nitpicks, to follow some git best practices
- directories and file names with spaces in them , e.g. "Azure DevOps" can be a bit cumbersome to work with locally
- azure pipelines maybe is the most appropriate name since azure devops includes more things than the CI/CD platform
- use all lowercase characters

suggestion = rename `Azure DevOps` to `azure_pipelines`, rename `ado_maven.yaml` to `azure_pipelines_maven.yml`
suggestion = make the other folders lowercase too e.g. `circleci` (use underscores instead of camel case where you want to)

**Comment by scott-es** [comment]
> > just FYI it looks like each pull request includes the commits from the prior one, what I'm saying is merging this one will bring in the 4 commits from PR 1-4 so once that happens I dont think the other 3 matter anymore.
> 
> @scott-es I was planning to rebase every PR, after the older one gets merged in. Wanted to keep changes limited in every PR.

since we want to just move a bunch of stuff over I don't see issue with doing at once, less work!

File: Jenkins/Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                sh 'node --version'
+
+                // Define the Node.js installation name configured in Jenkins
+                script {
+                    NODEJS_HOME = tool name: 'NodeJS', type: 'jenkins.plugins.nodejs.tools.NodeJSInstallation'
+                    PATH = "$NODEJS_HOME/bin:${env.PATH}"
+                }
+
+                // Download and install endorctl.
+                sh 'npm install -g endorctl'
```
Comment (scott-es) [inline Jenkins/Jenkinsfile]: I'm thinking we might prefer not to require `npm` to install `endorctl` in the pipeline.  can we replace with similar approach as the other examples to download the latest release and verify the checksum?  Then we remove the dependency on another package to be present and is consistent with the other examples

**Review by scott-es — APPROVED** [review]

File: Jenkins/Jenkinsfile (hunk 1 above)
Comment (ananthkamath) [reply Jenkins/Jenkinsfile]: I agree, I will make required changes.

Comment (ananthkamath) [reply Jenkins/Jenkinsfile]: @scott-es I have updated Jenkinsfile to use endorctl executable instead of npm package.

**Comment by ananthkamath** [comment]
> a few nitpicks, to follow some git best practices
> 
> * directories and file names with spaces in them , e.g. "Azure DevOps" can be a bit cumbersome to work with locally
> * azure pipelines maybe is the most appropriate name since azure devops includes more things than the CI/CD platform
> * use all lowercase characters
> 
> suggestion = rename `Azure DevOps` to `azure_pipelines`, rename `ado_maven.yaml` to `azure_pipelines_maven.yml` suggestion = make the other folders lowercase too e.g. `circleci` (use underscores instead of camel case where you want to)

@scott-es I have updated the folder names here, the files within Github and Gitlab pipelines can be named better. Will do that in the next iteration as follow up PR, as we need to sync files for these from the other repository.

**Review by scott-es — APPROVED** [review]
Merge!

**Pull request merged**